### PR TITLE
Update the test 'test_recovery_from_volume_deletion' to be compatible with vSphere LSO

### PR DESCRIPTION
**Pull request by yitzhak12**
1. Update the test 'test_recovery_from_volume_deletion' to be compatible with vSphere LSO
2. Add a new class in `platfrom_nodes` with the LSO functionality

### CODE REVIEW

**Review by ebenahar — COMMENTED** [review]
LGTM

File: ocs_ci/utility/vsphere.py [hunk 1]
```diff
@@ -1333,3 +1312,56 @@ def find_vms_without_ip_and_restart(self):
             logger.info(f"VM: {vm.name} doesn't have IP, it will be restarted!")
         if vms_without_ip:
             self.restart_vms(vms_without_ip, force=True)
+
+    def get_device_by_key(self, vm, identifier, key="unit_number"):
```
Comment (ebenahar) [inline ocs_ci/utility/vsphere.py:1316]: good idea to move this into a new method

Comment (yitzhak12) [reply ocs_ci/utility/vsphere.py:1316]: yes

File: tests/manage/z_cluster/nodes/test_disk_failures.py [hunk 2]
```diff
@@ -365,6 +368,12 @@ def test_recovery_from_volume_deletion(self, nodes, pvc_factory, pod_factory):
             osd_pv.delete()
             osd_pv.ocp.wait_for_delete(resource_name=osd_pv_name)
 
+        # If we use LSO, we need to create and attach a new disk manually
```
Comment (ebenahar) [inline tests/manage/z_cluster/nodes/test_disk_failures.py:371]: as discussed offline, this test is now adjusted to LSO VMDK.
Thinking we should have an issue to track the implementation of LSO PCI and LSO RDM

**Review by ebenahar — APPROVED** [review]

File: tests/manage/z_cluster/nodes/test_disk_failures.py (hunk 2 above)
Comment (yitzhak12) [reply tests/manage/z_cluster/nodes/test_disk_failures.py:371]: Create the issue https://github.com/red-hat-storage/ocs-ci/issues/3583

**Review by sidhant-agrawal — APPROVED** [review]
lgtm

**Pull request merged**